lib/application.rb

```diff
@@ -1,28 +1,18 @@
 class Application
-  def initialize(database_connection:, io:, order_repository:, item_repository:)
+  def initialize(io:, order_repository:, item_repository:)
     @io = io
     @order_repository = order_repository
-    @item_repository = ItemRepository.new(database_connection: @database_connection)
+    @item_repository = item_repository
   end
 
   def run
     @io.puts 'Welcome to the shop management program!'
-    loop do
-      menu
-      input = @io.gets.chomp
-      case input
-      when '1' 
-        print_all_items
-      when '2'
-        add_item
-      when '3'
-        print_all_orders
-      when '4'
-        add_order
-      when '5'
-        return
-      end
-    end
+    menu
+    input = @io.gets.chomp
+    print_all_items if input == '1'
+    add_item if input == '2'
+    print_all_orders if input == '3'
+    add_order if input == '4'
```

Comment:
Nice and short, I personally prefer using case, as it's even more readable at a glance.

Reply:
Yes I did use case at first as well, then the rubocop told me my method had too many lines, and I figured this to reduce number of lines. I agree that case is much more readable!